private/buf/bufformat/formatter.go

```diff
@@ -104,7 +114,22 @@ func (f *formatter) Indent() {
 
 // WriteString writes the given element to the generated output.
 func (f *formatter) WriteString(elem string) {
-	if _, err := fmt.Fprint(f.writer, elem); err != nil {
+	if f.pendingSpace {
```

Comment:
I added code to try to preserve some of the whitespace conventions in inline comments. That means it's possible for it to now call `f.Space()` but then the next call wants to write a newline. And trailing whitespace looks ugly.

So we now defer actually writing the space and skip it if the next character to write is a newline.